File: SofiForce.Models/Models/EntityModels/PromotionViewModel.cs
using System;

namespace Models
{
    public class PromotionViewModel
    {


        public string ConditionTypeID { get; set; }

        public int PromotionInput { get; set; }
        public int PromotionOutput { get; set; }

        public string PromotionInputName { get; set; }
        public string PromotionOutputName { get; set; }

        public string Description { get; set; }
        public int? PromotionCategoryId { get; set; }
        public string Prioritycode { get; set; }
        public Int32? Priorityvalue { get; set; }
        public string Promotionid { get; set; }
        public bool Promotionorderstatus { get; set; }
        public Int32? RepeatsNo { get; set; }
        public DateTime? Validfrom { get; set; }
        public DateTime? Validto { get; set; }


        public List<PromotionItemCritriaViewModel> PromotionItemCritria { get; set; } = new List<PromotionItemCritriaViewModel>();
        public List<PromotionMixAndMatchGroupViewModel> PromotionMixAndMatchGroup { get; set; } = new List<PromotionMixAndMatchGroupViewModel>();
        public List<PromotionOutcomeViewModel> PromotionOutcome { get; set; } = new List<PromotionOutcomeViewModel>();
        public List<PromotionClientViewModel> PromotionClient { get; set; } = new List<PromotionClientViewModel>();
        public List<PromotionHistoryViewModel> PromotionHistory { get; set; } = new List<PromotionHistoryViewModel>();

    }
    public class PromotionItemCritriaViewModel
    {
        public bool Exclude { get; set; }
        public Int32? GroupNo { get; set; }
        public string ItemCode { get; set; }
        public string ItemName { get; set; }

        public string PromotionId { get; set; }
        public double? GroupSlice { get; set; }

    }
    public class PromotionMixAndMatchGroupViewModel
    {
        public string PromotionId { get; set; }
        public Int32? GroupNo { get; set; }
        public double? Slice { get; set; }
    }
    public class PromotionOutcomeViewModel
    {
        public string PromotionId { get; set; }
        public Int32? Count { get; set; }
        public string ItemCode { get; set; }
        public string ItemName { get; set; }

        public Int32? Option { get; set; }  
        public double? Percentage { get; set; }
        public double? Quantity { get; set; }
        public double? Slice { get; set; }
        public double? Value { get; set; }
    }
    public class PromotionClientViewModel
    {
        public string PromotionId { set; get; }
        public string ClientCode { set; get; }
        public string ClientName { set; get; }

        public bool Exclude { set; get; }
    }
    public class PromotionHistoryViewModel
    {
        public string PromotionId { set; get; }
        public string SalesCode { set; get; }
        public string ClientCode { set; get; }
    }
}

File: SofiForce.Web/ClientApp/src/app/core/Models/EntityModels/PromotionViewModel.ts
export interface PromotionViewModel {
    conditionTypeID: string;
    promotionInput: number | null;
    promotionOutput: number | null;
    promotionInputName: number | null;
    promotionOutputName: number | null;
    description: string;
    PromotionCategoryId: number;
    prioritycode: string;
    priorityvalue: number | null;
    promotionid: string;
    promotionorderstatus: boolean | null;
    repeatsNo: number | null;
    validfrom: Date | null;
    validto: Date | null;
    promotionItemCritria: PromotionItemCritriaViewModel[];
    promotionMixAndMatchGroup: PromotionMixAndMatchGroupViewModel[];
    promotionOutcome: PromotionOutcomeViewModel[];
    promotionClient: PromotionClientViewModel[];
    promotionHistory: PromotionHistoryViewModel[];
}

export interface PromotionItemCritriaViewModel {
    exclude: boolean;
    groupNo: number | null;
    itemCode: string;
    itemName: string;
    promotionId: string;
    groupSlice:number;
}

export interface PromotionMixAndMatchGroupViewModel {
    promotionId: string;
    groupNo: number | null;
    slice: number | null;
}

export interface PromotionOutcomeViewModel {
    promotionId: string;
    count: number | null;
    itemCode: string;
    itemName: string;
    option: number | null;
    percentage: number | null;
    quantity: number | null;
    slice: number | null;
    value: number | null;
}

export interface PromotionClientViewModel {
    promotionId: string;
    clientCode: string;
    exclude: boolean;
}

export interface PromotionHistoryViewModel {
    promotionId: string;
    salesCode: string;
    clientCode: string;
}